src/vivarium/framework/resource/manager.py

```diff
@@ -189,20 +188,16 @@ def _to_graph(self) -> nx.DiGraph:
 
         return resource_graph
 
-    def __iter__(self) -> Iterator[Any]:
-        """Returns a dependency-sorted iterable of population initializers.
+    def get_population_initializers(self) -> list[Any]:
+        """Returns a dependency-sorted list of population initializers.
 
         We exclude all non-initializer dependencies. They were necessary in
         graph construction, but we only need the column producers at population
         creation time.
         """
-        return iter(
-            [
-                r.producer
-                for r in self.sorted_nodes
-                if r.type in {"column", NULL_RESOURCE_TYPE}
-            ]
-        )
+        return [
```

Comment:
Should we be worried about order?

Reply:
This change doesn't affect the order in which the initializers will be returned.